machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,112 @@
+
+	machineosconfig "github.com/openshift/api/machineconfiguration/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=internalreleaseimages,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2510
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=NoRegistryClusterOperations
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+
+// InternalReleaseImage is used to keep track and manage a set
+// of release bundles (OCP and OLM operators images) that are stored
+// into the control planes nodes.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type InternalReleaseImage struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this internal release image.
+	// +required
+	Spec InternalReleaseImageSpec `json:"spec,omitzero"`
+
+	// status describes the last observed state of this internal release image.
+	// +optional
+	Status *InternalReleaseImageStatus `json:"status,omitempty"`
+}
+
+// InternalReleaseImageStatus describes the current state of a InternalReleaseImage.
+type InternalReleaseImageStatus struct {
+	// conditions represent the observations of an internal release image current state.
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=256
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// availableReleases is a list of release bundle identifiers currently detected
+	// from the attached ISO.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=64
+	// +optional
+	AvailableReleases []InternalReleaseImageRef `json:"availableReleases,omitempty"`
+
+	// releases is a list of the currently managed release bundles.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=64
+	// +optional
+	Releases []InternalReleaseImageRef `json:"releases,omitempty"`
+}
+
+// InternalReleaseImageSpec defines the desired state of a InternalReleaseImage.
+type InternalReleaseImageSpec struct {
+	// releases is a list of release bundle identifiers that the user wants to
+	// add/remove to/from the control plane nodes.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=20
```

Comment:
Same comment as above. I assume we'd always have the spec and status map 1-to-1, so maybe they should have the same limits?